R/filterReadsBam.R

```diff
@@ -0,0 +1,174 @@
+#' Write records from \code{infile} to \code{outfile} if they pass filter criteria.
+#'
+#' For each file in \code{infiles}, parse alignments, calculate read statistics
+#' and write the alignment to the corresponding output file from \code{outfiles}
+#' if the read passes all criteria defined by the filtering arguments.
+#' Filters are processed hierarchically: If a read does not pass a given
+#' filter, the remaining filters will not be examined and the processing
+#' continues with the next read.
+#' The filters are examined in this order: \code{minReadLength},
+#' \code{minAlignedLength}, \code{minAlignedFraction}, \code{minQscore},
+#' \code{maxFracLowConf}, \code{maxEntropy}.
+#'
+#' @param infiles Character vector with name(s) of the input bam file(s).
+#'     If \code{infiles} is a named vector, the names are used as sample names.
+#'     Otherwise, the each file will be names as \code{s1}, ..., \code{sN},
+#'     where \code{N} is the length of \code{infiles}.
+#' @param outfiles Character vector with name(s) of the output bam file(s).
+#'     Needs to have the same length as \code{infiles}.
+#' @param modbase Character scalar defining the modified base to analyze
+#'     (used by \code{maxEntropy} and \code{maxFracLowConf}).
+#' @param indexOutfiles Logical scalar. If \code{TRUE} (the default) create
+#'     a bam index file (\code{.bai} file) for each of the generated
+#'     \code{outfiles}.
+#' @param minReadLength A numeric scalar representing the smallest acceptable
+#'     read length. Reads that are shorter than this value will be filtered
+#'     out.
+#' @param minAlignedLength A numeric scalar representing the smallest acceptable
+#'     aligned length. Reads with aligned length shorter than this value will
+#'     be filtered out.
+#' @param minAlignedFraction A numeric scalar representing the smallest
```

Comment:
Just wondering if we should inherit parameter documentation between filterReads and filterReadsBam to avoid having to define them twice.

Reply:
Yes, I was also thinking that would be preferable. Would you combine the man pages of `filterReads` and `filterReadsBam` (similarly as for `calcReadStats` and `addReadStats`), or do you think there would be another way to do this (e.g. just removing the details from the man page of `filterReadsBam` and instead add a link to the man page of `filterReads`)?

Reply:
Good question - we could either combine the pages, or add 

```
#' @inheritParams filterReads
```

to `filterReadsBam`, in which case it will inherit all shared arguments.

Reply:
Thanks for the hint - the `@inherit` keywords are amazing! Implemented in c991e5e8